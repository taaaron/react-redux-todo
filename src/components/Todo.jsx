import React from 'react';
import PropTypes from 'prop-types';

  const Todo=(props)=> {

    return (
        <div>
            <p>
                {props.todo.text}
                <input type='checkbox'/>
            </p>
        </div>)
}

Todo.propTypes = {
    todo: PropTypes.object.isRequired
}
export default Todo;

