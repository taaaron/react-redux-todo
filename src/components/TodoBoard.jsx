import React from 'react';
import Todo from './Todo';
import {connect} from 'react-redux';

export class TodoBoard extends React.Component{

    render(){
        console.log(this.props);

        return(
            <div>
                {
                    this.props.todos.map(todo => {
                        return <Todo key='1' todo={todo}/>
                    })
                }
            </div>
        )
    }

}

const mapStateToProps = (state) => {
    return {
        todos: state.todos
    }
}

export default connect(mapStateToProps)(TodoBoard);




