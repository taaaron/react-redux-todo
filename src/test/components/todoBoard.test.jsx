import {shallow, mount} from 'enzyme';
import {TodoBoard} from '../../components/TodoBoard';
import React from 'react';
import Todo from '../../model/todo';

describe('todoBoard should display todos',()=> {
    it('should have 1 todo displayed',()=>{
        const initialState = {
            todos: [new Todo("Friday afternoon stuff with A-Aaron & Manny")]
        };

        const wrapper = shallow(<TodoBoard todos={initialState.todos}/>);

         expect(wrapper.find('Todo')).toHaveLength(1);
    })
})