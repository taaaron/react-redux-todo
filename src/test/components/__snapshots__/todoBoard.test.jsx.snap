// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`todoBoard should display todos should have 1 todo displayed 1`] = `
<div>
  <div>
    <p>
      props.todo
      <input
        type="checkbox"
      />
    </p>
  </div>
</div>
`;
